### PR TITLE
StateByCategoryPrefix RPC

### PR DESCRIPTION
**Pull request by maxtaco**
- Also some code cleanups as I saw issues.

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @mmaxim 

Also @patrickxb there are a few cleanups here

File: rpc/server/alive_group.go [hunk 1]
```diff
@@ -66,15 +66,15 @@ func (a *aliveGroup) Publish(ctx context.Context, msg gregor1.Message) error {
...
+		go func(id srvup.NodeId, conn *sibConn) {
 			if err := conn.CallConsumePublishMessage(ctx, msg); err != nil {
 				a.log.Warning("consumePubMessage error: id: %s addr: %s err: %s", id, conn.address, err)
 				perr.Add(conn.address, err)
 			} else {
 				a.log.Debug("consumePubMessage success: id: %s addr: %s", id, conn.address)
 			}
 			wg.Done()
-		}()
+		}(id, conn)
```
Comment (mmaxim) [inline rpc/server/alive_group.go]: Wow I didn't even notice that, good find!

**Comment by mmaxim** [comment]
LGTM!

File: rpc/server/alive_group.go (hunk 1 above)
Comment (maxtaco) [inline rpc/server/alive_group.go]: Go vet found it!

**Pull request merged**